src/utils.ts

```diff
@@ -574,3 +575,58 @@ export async function approveERC20(
   const erc20 = BaseErc20Factory.connect(erc20Address, wallet)
   return erc20.approve(spender, amount)
 }
+
+/**
+ * Returns the `verified` status of a uri.
+ * A uri is only considered `verified` if its content hashes to its expected hash
+ *
+ * @param uri
+ * @param expectedHash
+ * @param timeout
+ */
+export async function isURIHashVerified(
+  uri: string,
+  expectedHash: BytesLike,
+  timeout: number = 10
+): Promise<boolean> {
+  try {
+    validateURI(uri)
+
+    const resp = await axios.get(uri, {
+      timeout: timeout,
+      responseType: 'arraybuffer',
+    })
+    const uriHash = sha256FromBuffer(resp.data)
+    const normalizedExpectedHash = hexlify(expectedHash)
+
+    return Promise.resolve(uriHash == normalizedExpectedHash)
```

Comment:
you don't need Promise.resolve here—since it's an async function returned values are wrapped in a promise